src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -659,14 +649,21 @@ public sealed override ImmutableArray<FieldSymbol> TupleElements
                                 else
                                 {
                                     Debug.Assert(fieldSymbol.Name == defaultName, "top level underlying field must match default name");
-
-                                    // Add the underlying/real field as an element (wrapping mainly to capture location). It should have the default name.
-                                    defaultTupleField = new TupleElementFieldSymbol(this,
-                                                                                    fieldSymbol,
-                                                                                    tupleFieldIndex,
-                                                                                    locations,
-                                                                                    isImplicitlyDeclared: defaultImplicitlyDeclared,
-                                                                                    correspondingDefaultFieldOpt: null);
+                                    if (IsDefinition)
+                                    {
+                                        defaultTupleField = field;
+                                        unwrappedFieldsToIndexMap.Add(field, tupleFieldIndex);
+                                    }
+                                    else
+                                    {
+                                        // Add the underlying/real field as an element (wrapping mainly to capture location). It should have the default name.
+                                        defaultTupleField = new TupleElementFieldSymbol(this,
+                                                                                        fieldSymbol,
+                                                                                        tupleFieldIndex,
+                                                                                        locations,
+                                                                                        isImplicitlyDeclared: defaultImplicitlyDeclared);
+                                        unwrappedFieldsToIndexMap.Add(fieldSymbol, tupleFieldIndex);
```

Comment:
>unwrappedFieldsToIndexMap.Add(fieldSymbol, tupleFieldIndex); [](start = 40, length = 60)

Is this going to do the right thing for a substituted field? What does ```defaultTupleField``` mean then? Is ```fieldSymbol``` going to be in the members list? If not, then why are we adding it to the map?